state.go

```diff
@@ -557,6 +560,117 @@ func (s *State) Channel(channelID string) (*Channel, error) {
 	return nil, ErrStateNotFound
 }
 
+// ThreadMemberUpdate updates the ThreadMember object for the user,
+// in a specific thread.
+func (s *State) ThreadMemberUpdate(t *ThreadMember) error {
+	if s == nil {
+		return ErrNilState
+	}
+
+	channel, ok := s.channelMap[t.ID]
+	if ok {
+		s.Lock()
+		defer s.Unlock()
+
+		channel.Member = t
+		return nil
+	}
+
+	return ErrStateNotFound
+}
+
+// ThreadMembersUpdate updates the member object of the thread.
+// If the user was removed,  the ThreadMember object in the state
+// will be set to nil.
+func (s *State) ThreadMembersUpdate(t *ThreadMembersUpdate) error {
+	if s == nil {
+		return ErrNilState
+	}
+
+	s.Lock()
+	defer s.Unlock()
+
+	thread, ok := s.channelMap[t.ID]
+	if ok {
+		for _, memberID := range t.RemovedMembersIDs {
+			if memberID == s.User.ID {
+				thread.Member = nil
+				break
+			}
+		}
+
+		for _, member := range t.AddedMembers {
+			if member.UserID == s.User.ID {
+				thread.Member = member
+				break
+			}
+		}
+	}
+
+	return ErrStateNotFound
```

Comment:
It seems to me like this function always returns `ErrStateNotFound` if it makes it past the initial `s == nil` check. I'd probably invert the `if` statement to `if !ok` to return early and keep the happy path left (in addition to above in `ThreadMemberUpdate`).

Reply:
Updated